medicines/web/src/services/keyword.ts

```diff
@@ -0,0 +1,38 @@
+const searchExactnessBoost = process.env.AZURE_SEARCH_EXACTNESS_BOOST;
+const searchWordFuzziness = process.env.AZURE_SEARCH_WORD_FUZZINESS;
+
+const extractProductLicenseRegExp: RegExp = new RegExp(
+  '(\\b|PL)(\\s+|/|_|-)*(\\d{5})(\\s+|/|_|-)*(\\d{4})',
+  'ig',
+);
+
+const escapeSpecialWords = (word: string): string =>
+  word.replace(/(\|\||&&|\bAND\b|\bOR\b|\bNOT\b)/gi, `\\$1`);
+
+const preferExactMatchButSupportFuzzyMatch = (word: string): string =>
+  `${word}~${searchWordFuzziness} ${word}^${searchExactnessBoost}`;
+
+const addNormalizedProductLicenses = (q: string): string => {
```

Comment:
What is `q`?

Reply:
This is a direct move of the code from https://github.com/MHRA/products/commit/8024817194e579bd284146de01e4ee4b1eedb7f4